server/services/referral.service.js

```diff
@@ -364,3 +364,38 @@ export const updateReferralRewardStatus = async ({ referralId, offerId }) => {
 
 export const getReferralByOfferId = async (offerId) =>
   Referral.findOne({ offerId: ObjectId(offerId) });
+
+export const updateAssignedReferral = async ({ referralId, transactionId, amountPaid }) => {
+  const referral = await getReferralBasicInfoById(referralId).catch((error) => {
+    throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
+  });
+
+  const user = await getUserById(referral.referrerId);
+
+  const percentage = user.additionalInfo.referralPercentage;
+  const amountPayable = (percentage / 100) * amountPaid;
+  const total =
+    amountPayable +
+    referral.assignedReferral.transactions.reduce(
+      (accum, transaction) => accum + transaction.amountPayable,
```

Comment:
Not sure why you used `accum`,  `acc` is a better alternative in this case.

```suggestion
      (acc, transaction) => acc + transaction.amountPayable,
```

Also, can we assign the reduce function to another variable like `previousTotal` to make it more understandable